rosidl_generator_c/test/test_interfaces.c

```diff
@@ -333,8 +337,39 @@ int test_strings(void)
   return 0;
 }
 
-
 /**
+ * Test message with different wstring types.
+ */
+int test_wstrings(void)
+{
+  bool res = false;
+  rosidl_generator_c__msg__WStrings * wstrings = NULL;
+
+  wstrings = rosidl_generator_c__msg__WStrings__create();
+  EXPECT_NE(wstrings, NULL);
+
+  res = rosidl_generator_c__U16String__assign(&wstrings->empty_wstring, TEST_WSTRING);
+  EXPECT_EQ(true, res);
+  // EXPECT_EQ(0, strcmp(wstrings->empty_wstring.data, TEST_WSTRING));
+  // EXPECT_EQ(0, strcmp(wstrings->def_wstring.data, "Hello world!"));
+  // EXPECT_EQ(0, strcmp(wstrings->def_wstring2.data, "Hello'world!"));
+  // EXPECT_EQ(0, strcmp(wstrings->def_wstring3.data, "Hello\"world!"));
+  // EXPECT_EQ(0, strcmp(wstrings->def_wstring4.data, "Hello'world!"));
+  // EXPECT_EQ(0, strcmp(wstrings->def_wstring5.data, "Hello\"world!"));
+  // since upper-bound checking is not implemented yet, we restrict the string copying
+  // TODO(mikaelarguedas) Test string length properly instead of cheating copy
```

Comment:
dirk-thomas

Reply:
1-to-1 copy from the `String` tests above. So I will keep it as is.